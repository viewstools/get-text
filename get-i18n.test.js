const getI18n = require('./get-i18n.js')

const READ_VIEW = `Read Vertical
Text
color #323232
fontFamily SlatePro
fontSize 20
fontWeight 600
text Assignment:

Name Text
text <name gets a default value

CaptureText
placeholder hi`

const READ_VIEW_THAT_ALREADY_HAS_TRANSLATIONS = `Read Vertical
Text
color #323232
fontFamily SlatePro
fontSize 20
fontWeight 600
text Assignment:
when <es
text Tarea:

NameLabel Text
text Name:
when <es
text Nombre:
when <deCH
text Namen:
`

test('#getI18n', () => {
  expect(getI18n('Read', READ_VIEW)).toMatchSnapshot()

  expect(
    getI18n('Read', READ_VIEW_THAT_ALREADY_HAS_TRANSLATIONS)
  ).toMatchSnapshot()
})
