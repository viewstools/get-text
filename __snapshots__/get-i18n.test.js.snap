// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#getI18n 1`] = `
Object {
  "default": Object {
    "Read/CaptureText/placeholder hi": "hi",
    "Read/Name Text/text <name gets a default value": "gets a default value",
    "Read/Text/text Assignment:": "Assignment:",
  },
}
`;

exports[`#getI18n 2`] = `
Object {
  "deCH": Object {
    "Read/NameLabel Text/text Name:": "Namen:",
  },
  "default": Object {
    "Read/NameLabel Text/text Name:": "Name:",
    "Read/Text/text Assignment:": "Assignment:",
  },
  "es": Object {
    "Read/NameLabel Text/text Name:": "Nombre:",
    "Read/Text/text Assignment:": "Tarea:",
  },
}
`;
